page-tracking/src/page_info.rs

```diff
@@ -28,6 +28,13 @@ pub enum PageState {
     /// Page is mapped into the address space of the current owner.
     Mapped,
 
+    /// Page is mapped into the address space of the current owner, but has been shared with
+    /// a child-VM. Shared pages cannot be converted
+    Shared,
```

Comment:
Why should the TSM need to track the sharing with other guests - once a page is converted into a non-confidential page, it is under the VMMs purview to share as needed; for the page to be repurposed, it will have to be converted to a confidential (unassigned) page and at that point the exclusive property can be enforced.

Reply:
> for the page to be repurposed, it will have to be converted to a confidential (unassigned) page and at that point the exclusive property can be enforced.

That is exactly the point of reference counting how many guests have mapped the shared page. Otherwise, how does the TSM, upon receiving a request to convert a page to confidential memory, enforce that the page is no longer shared?